### DISCUSSION TITLE
更新日志 - 22.03.5 - 20230928

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1022] [Announcements]
<b>x86 系列机型有较大更新</b>

# 系统
1. 增加内存压缩功能（ZRam），默认关闭，可在后台页面“系统”-“系统”的“ZRam 设置”开启此功能（下次启动生效）
2. 集成 USB 声卡和摄像头驱动

# 设备
* x86：
  1. 移植 Intel GPU 厂商版驱动（i915-oot），理论上可支持最新的 Intel MeteorLake/14代 CPU 的核显以及独显，默认未启用（未启用情况下使用内核提供的 i915 驱动），只建议11代以上的 CPU 启用，老 CPU 启用可能会有问题，启用方法见下文
  2. i915 驱动增加更多 GPU 固件，以支持10代及以下 CPU 核显（例如 J4125）
  3. 集成 Realtek PCIe 网卡驱动 r8125 和 r8168，默认未启用，启用方法见下文。如果之前手动安装过 r8125 或 r8168 驱动，建议先卸载掉驱动再升级固件，升级以后再按下文的方式启用驱动

# x86 额外驱动的使用说明
如果你的 CPU 是 N5095 及以上，可以启用 i915-oot，否则不建议启用。

如果遇到 Realtek PCIe 网卡问题，例如虚拟机内不能正常使用，网速不达标等，如果是 RTL8125 网卡可以尝试启用 r8125 驱动，其他网卡尝试启用 r8168 驱动。

额外驱动启用方法：在后台页面“系统”-“杂项设置”的“驱动”页签里，勾选需要的驱动并保存应用，将在下次启动时生效。

**Comment by sileng** [comment]
8156B无法识别2.5G速率

**Reply by jjm2473** [reply]
什么硬件？如果确定是bug，开个issue。

**Comment by rayangl** [comment]
更新系统风扇一直狂转，设备是r2s,然后一直不启动,需要拔电源再插电源才能启动,他这是一种概率事件，我设置了每周重启一次，他有时候能重启成功，有时候他就需要拔掉电源再重启,感觉是卡死了，两个网口的灯都不亮，系统的灯也不亮,就风扇一直在狂转

**Reply by xpn282** [reply]
这是r2s的特色了，我朋友的就是这样，概率性无法重启，必须拔电源。固件也换了，电源也换了，都无法解决，我觉得是硬件问题，他用得生不如死

**Reply by Vonation** [reply]
8月份更新时就出现过类似情况，后来回档之前版本，重新写卡不再更新了

**Reply by jjm2473** [reply]
r2s好像有概率识别不到tf卡，用老的tf卡可能就没问题了。

**Comment by Mr7eft** [comment]
更新至最新版本。N100，用ESXI8.0b，显卡直通ISTOREOS，已在杂项设置，勾选移植 Intel GPU 驱动 (i915-oot)，并重启。
jellyfin勾选硬解后，cpu占用率还是很高，请教一下是什么原因？

**Reply by jjm2473** [reply]
是不是qsv？勾选了所有解码器吗？勾选了低电压编码h264这些选项吗？如果勾选了，还能转码，那肯定是硬解了。至于CPU的使用率，就不清楚了，可能是硬盘的IO有问题？

**Comment by WarlockFish** [comment]
更新至最新版本后，运行zerotier后机器就会自动重启

**Reply by WarlockFish** [reply]
测试发现2023090810版本启动zerotier正常，之后版本启动zerotier机器就重启，日志也没记录下来，配置rsyslog远程也无日志

**Reply by jjm2473** [reply]
是否有打开额外驱动？`lspci -nnk`看看驱动对不对

**Reply by WarlockFish** [reply]
我是重新安装全新系统，zerotier会用到驱动吗？我使用的是pve的virtio,测试其他网卡也是这种现象

**Reply by jjm2473** [reply]
zerotier会用到wireguard驱动，网络的话肯定会经过网卡啊。我看看是不是wireguard的驱动没重新编译

**Reply by jjm2473** [reply]
我看了下，wireguard的驱动应该没问题，是跟着内核编译的。你可以给虚拟机增加一个com口（或者叫串口），在pve中监视此串口的输出，然后启动istoreos，在grub启动阶段选择“iStoreOS (ttyS0)”，就可以在pve的串口终端看到内核日志了，然后开启zerotier看看有没有抓到日志

**Reply by jjm2473** [reply]
哦，zerotier不是使用wireguard驱动，而是tun驱动。我刚测试了zerotier，连接到zerotier是没问题的。
<img width="807" alt="image" src="https://github.com/istoreos/istoreos/assets/3896703/a6641e30-8a10-4931-928f-534a175e41ae">

**Reply by WarlockFish** [reply]
不行，没有日志。
我启动iStoreOS (ttyS0)，开启zerotier后能开机1分钟，能看见zerotier的ip，1分钟后就机器就重启了，再次开机就无限重启，只能进入单用户把zerotier的开机自启配置关闭后，机器才能拉起来。
![图片](https://github.com/istoreos/istoreos/assets/30708115/c9f09b46-02ed-4513-9a97-ddb87fda5299)

开启zerotier后，iStoreOS正常模式直接重启。
![截图 2023-10-06 20-10-05](https://github.com/istoreos/istoreos/assets/30708115/b685d024-624c-477a-83dc-e0c48458ebc6)

**Reply by jjm2473** [reply]
这个报错说明你的系统文件损坏了，估计是你更新了系统底层的库，procd都报错了

**Reply by WarlockFish** [reply]
新装的启动zerotier后也重启。procd只是重启第一次报出来的，后面就无限重启。还没到这就重启了。

**Reply by jjm2473** [reply]
是不是你连的zerotier网络里面有病毒什么的，把系统给破坏了

**Comment by teddying1000** [comment]
n4500,Jellyfin硬解转码就报错，不支持的媒体之类的

root@iOS:~# grep -F 'model name' /proc/cpuinfo | head -1; lspci -nn | grep -F VGA; dmesg | grep -F i915
model name      : Intel(R) Celeron(R) N4500 @ 1.10GHz
00:02.0 VGA compatible controller [0300]: Intel Corporation JasperLake [UHD Graphics] [8086:4e55] (rev 01)
[    7.202722] enable_kmod.group i915-oot
[    8.224295] i915 0000:00:02.0: [drm] GT count: 1, enabled: 1
[    8.224646] i915 0000:00:02.0: [drm] Unsupported device. GVT-g is disabled
[    8.224715] i915 0000:00:02.0: [drm] Transparent Hugepage support is recommended for optimal performance on this platform!
[    8.244570] i915 0000:00:02.0: [drm] Finished loading DMC firmware i915/icl_dmc_ver1_09.bin (v1.9)
[    8.907628] i915 0000:00:02.0: [drm] [ENCODER:238:DDI A/PHY A] failed to retrieve link info, disabling eDP
[    8.935083] i915 0000:00:02.0: [drm] GT0: GuC firmware i915/ehl_guc_70.7.0.bin version 70.7.0
[    8.935092] i915 0000:00:02.0: [drm] GT0: HuC firmware i915/ehl_huc_9.0.0.bin version 9.0.0
[    8.952177] i915 0000:00:02.0: [drm] GT0: HuC: authenticated!
[    8.953281] i915 0000:00:02.0: [drm] GT0: GUC: submission enabled
[    8.953318] i915 0000:00:02.0: [drm] GT0: GUC: SLPC disabled
[    8.955212] [drm] Initialized i915 1.6.0 20201103 for 0000:00:02.0 on minor 0
[   10.307896] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[   11.649464] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[   13.418820] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[   88.202615] i915 0000:00:02.0: [drm] *ERROR* GT0: GUC: Engine reset request failed on 0:0 (rcs0) because 0x0, GDRST = 0x00000002
[   88.203142] i915 0000:00:02.0: [drm] *ERROR* GT0: GUC: i915 wait for GDRST also failed: -110 [on 0:0 (rcs0)]
[   88.456879] i915 0000:00:02.0: GPU HANG: ecode 11:1:8ed7fff1, rcs0 in ffmpeg [18618]
[   88.456934] i915 0000:00:02.0: [drm] Resetting chip for GuC failed to reset rcs0 (reason=0x00000000)
[   88.459010] i915 0000:00:02.0: [drm] *ERROR* GT0: Failed to reset GuC, ret = -110
[   88.692832] i915 0000:00:02.0: [drm] *ERROR* GT0 [GT OTHER] Failed to reset chip
[   88.692920] i915 0000:00:02.0: [drm:add_taint_for_CI [i915]] CI tainted:0x9 by intel_gt_reset+0x1b9/0x4b0 [i915]
[   88.694991] i915 0000:00:02.0: [drm] *ERROR* GT0: Failed to reset GuC, ret = -110

![image](https://github.com/istoreos/istoreos/assets/110078018/f981c671-1571-4259-bca4-b4e6e9e3f2a2)
![image](https://github.com/istoreos/istoreos/assets/110078018/ebfc53bc-b78c-4349-9a89-9e146edc9f45)
![image](https://github.com/istoreos/istoreos/assets/110078018/f07f3706-cfa7-4bc3-82a4-31dbdf836045)

**Reply by jjm2473** [reply]
那就别启用啊，不是说了N5095以上才可以用这个驱动吗？老的CPU用默认的驱动就行了，默认驱动都支持。

**Reply by teddying1000** [reply]
就是因为不启用的时候，硬件转码报播放错误，刚好更新了固件，才试了勾选这个，还是不行。
以下是不勾选应用重启后的信息
root@iOS:~# grep -F 'model name' /proc/cpuinfo | head -1; lspci -nn | grep -F VGA; dmesg | grep -F i915
model name      : Intel(R) Celeron(R) N4500 @ 1.10GHz
00:02.0 VGA compatible controller [0300]: Intel Corporation JasperLake [UHD Graphics] [8086:4e55] (rev 01)
[    7.167382] disable_kmod.group i915-oot
[    7.764498] i915 0000:00:02.0: [drm] Incompatible option enable_guc=3 - GuC submission is N/A
[    7.776238] i915 0000:00:02.0: [drm] Finished loading DMC firmware i915/icl_dmc_ver1_09.bin (v1.9)
[    8.439595] i915 0000:00:02.0: [drm] failed to retrieve link info, disabling eDP
[    8.479779] i915 0000:00:02.0: [drm] GuC firmware i915/ehl_guc_33.0.4.bin version 33.0 submission:disabled
[    8.479788] i915 0000:00:02.0: [drm] HuC firmware i915/ehl_huc_9.0.0.bin version 9.0 authenticated:yes
[    8.481665] [drm] Initialized i915 1.6.0 20200917 for 0000:00:02.0 on minor 0
[    9.152486] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[    9.890163] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[   10.560178] i915 0000:00:02.0: [drm] Cannot find any crtc or sizes
[   78.680278] i915 0000:00:02.0: [drm] Resetting rcs0 for preemption time out
[   78.681004] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   78.904295] i915 0000:00:02.0: [drm] GPU HANG: ecode 11:1:8ed7fff0, in ffmpeg [19124]
[   90.163467] i915 0000:00:02.0: [drm] GPU HANG: ecode 11:1:8ed7fff0, in ffmpeg [19124]
[   90.265169] i915 0000:00:02.0: [drm] Resetting rcs0 for stopped heartbeat on rcs0
[   90.265883] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.265955] i915 0000:00:02.0: [drm] Resetting chip for stopped heartbeat on rcs0
[   90.368652] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.369362] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.372138] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.403282] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.404006] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.406786] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.443203] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.443927] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.446706] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.493129] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.493852] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.496633] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.498707] i915 0000:00:02.0: [drm] *ERROR* Failed to reset chip
[   90.498777] i915 0000:00:02.0: [drm:add_taint_for_CI [i915]] CI tainted:0x9 by intel_gt_reset+0x2a1/0x460 [i915]
[   90.602393] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.603116] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.605895] i915 0000:00:02.0: [drm] *ERROR* rcs0 reset request timed out: {request: 00000001, RESET_CTL: 00000001}
[   90.608002] i915 0000:00:02.0: [drm] ffmpeg[19124] context reset due to GPU hang
[   90.627680] i915 0000:00:02.0: [drm] *ERROR* rcs0 TLB invalidation did not complete in 4ms!
[   90.632941] i915 0000:00:02.0: [drm] *ERROR* bcs0 TLB invalidation did not complete in 4ms!
[   90.638408] i915 0000:00:02.0: [drm] *ERROR* vcs0 TLB invalidation did not complete in 4ms!
[   90.643832] i915 0000:00:02.0: [drm] *ERROR* vecs0 TLB invalidation did not complete in 4ms!
[   90.649423] i915 0000:00:02.0: [drm] *ERROR* rcs0 TLB invalidation did not complete in 4ms!
[   90.654468] i915 0000:00:02.0: [drm] *ERROR* bcs0 TLB invalidation did not complete in 4ms!
[   90.659800] i915 0000:00:02.0: [drm] *ERROR* vcs0 TLB invalidation did not complete in 4ms!
[   90.664587] i915 0000:00:02.0: [drm] *ERROR* vecs0 TLB invalidation did not complete in 4ms!
[   90.669912] i915 0000:00:02.0: [drm] *ERROR* rcs0 TLB invalidation did not complete in 4ms!
[   90.674956] i915 0000:00:02.0: [drm] *ERROR* bcs0 TLB invalidation did not complete in 4ms!

**Reply by jjm2473** [reply]
你在jellyfin里面不勾选“低电压模式”试试

**Reply by jjm2473** [reply]
还有把“首选系统原生的DXVA”什么的选上看看

**Comment by jjm2473** [comment]
注意一下，N5095以下的老CPU不需要启用i915-oot驱动，默认的i915驱动就支持硬解了，启用i915-oot反而可能导致各种问题。

**Reply by wsmayi** [reply]
这个i915-oot驱动的源码可以共享一下吗？想试试看能不能移植到黑群晖SA6400的DSM 7.2(内核版本为5.10.55)上

**Reply by jjm2473** [reply]
@wsmayi  istoreos都是开源的啊，这是相关的makefile和源码补丁  https://github.com/istoreos/istoreos/tree/istoreos-22.03/package/kernel/i915-oot

**Comment by xqk1** [comment]
i915-oot 相比于 i915 有什么差异么，网上搜不太到相关信息

**Reply by jjm2473** [reply]
支持新的intel gpu

**Comment by paidaxinghappy** [comment]
arm版有啥提升？

**Reply by jjm2473** [reply]
没标记x86的变更就是共同的

**Comment by ArnoChenFx** [comment]
更新至最新版本。N100，用ESXI8.0.0，显卡直通ISTOREOS，已在杂项设置，勾选"移植移植 Intel GPU 驱动 (i915-oot)"，并重启。
jellyfin无法硬件解码。
![image](https://github.com/istoreos/istoreos/assets/24894259/9a00cf38-59f5-4177-9586-c7f3da339a01)
![image](https://github.com/istoreos/istoreos/assets/24894259/d047952c-1e62-4277-a005-44587ffd745d)

**Reply by ArnoChenFx** [reply]
![image](https://github.com/istoreos/istoreos/assets/24894259/46176a79-ffaa-41fc-be27-8064ba95ed52)
![image](https://github.com/istoreos/istoreos/assets/24894259/c721198b-081b-4877-bad9-cfe7854b5b5a)

**Reply by jjm2473** [reply]
没测试过esxi，可能有其他问题。你有遇到其他固件解决了esxi虚拟机的intel核显的吗，提供参考下

**Reply by ArnoChenFx** [reply]
在论坛找到有人用iStoreOS测试固件成功过
https://www.right.com.cn/forum/thread-8307024-1-1.html

**Reply by skld315** [reply]
我是没用虚拟机，直接物理机跑，硬解HDR 就有问题，CPU N100

**Reply by ArnoChenFx** [reply]
Jellyfin日志中的报错如下：
[AVHWDeviceContext @ 0x55f1bba29940] No VA display found for any default device.
Device creation failed: -22.
Failed to set value 'vaapi=va:,driver=iHD,kernel_driver=i915' for option 'init_hw_device': Invalid argument
Error parsing global options: Invalid argument

**Reply by jjm2473** [reply]
@ArnoChenFx 你的jellyfin是自己用命令部署的还是用istore里面的jellyfin插件部署的？

如果使用插件部署的，是否修改过默认镜像？

**Reply by ArnoChenFx** [reply]
使用的是iStore中的jellyfin

**Reply by s949492225** [reply]
我也是物理机装的N100，也遇到这个问题，我已经提了一个bug了

**Reply by ArnoChenFx** [reply]
最新的22.03.5 - 20231027固件已经解决这个问题了

**Reply by weiqincc** [reply]
我从测试版固件 升级到最新的  iStoreOS 22.03.5 2023120811 出现了和你一样的问题，怎么解决的哦，请求大佬支援

**Comment by Sutong888** [comment]
您好我是X86，螃蟹8111网卡，按照您提示去添加r8168驱动，显示保存设置成功后无法打开后台了，并且电脑也提取不到默认网关，这个如何解决呢

**Reply by Sutong888** [reply]
至于为啥添加驱动，因为带宽才跑一半，只有500-600兆，无法跑满千兆

**Reply by jjm2473** [reply]
看看显示器输出什么

**Reply by jjm2473** [reply]
有没有报错什么的

**Reply by jjm2473** [reply]
“显示保存设置成功后无法打开后台了”？这怎么可能，驱动的修改要下次开机才有效。我怀疑你自己改了lan口的ip，但是没保存应用，然后在这个插件保存应用就把lan口ip也应用了。

**Comment by skld315** [comment]
> 我是没用虚拟机，直接物理机跑，硬解HDR 就有问题，CPU N100
更新：我发现不一定是视频解码问题，音频解码也会导致，在这个电影中只有粤语音轨能正常播放，只要切换到别的音轨就会报错
![image](https://github.com/istoreos/istoreos/assets/45231589/91081af2-70d4-43df-acdf-61ff91163049)
![image](https://github.com/istoreos/istoreos/assets/45231589/73052624-0d76-4217-9d9b-c8dabf63613e)
![image](https://github.com/istoreos/istoreos/assets/45231589/5e22168e-92b1-4ebf-919e-3e3f1b045ba7)

**Reply by jjm2473** [reply]
那估计是jellyfin的bug，看看jellyfin的版本

**Reply by skld315** [reply]
在镜像里用jellyfin/jellyfin:latest 拉的
![image](https://github.com/istoreos/istoreos/assets/45231589/89b972fc-307c-4f92-b0f5-65223a20921f)

**Reply by jjm2473** [reply]
试试上一个版本10.8.10

**Reply by skld315** [reply]
换回上个版本，问题解决！！，就是配置docker整了一会，谢谢

**Reply by jjm2473** [reply]
肯定跟固件版本无关，你说的问题是音频转码，甚至跟GPU驱动都无关。

**Reply by jjm2473** [reply]
jellyfin有转码日志，以后出问题应该看看转码日志，在jellyfin的控制台-日志那里，找transcode有关的日志。

**Reply by skld315** [reply]
好的，再次感谢

**Comment by HongminTan** [comment]
您好，我是pve8下安装的istoreos，固件为最新（22.03.5），利用n100的VT-d功能虚拟了gpu出来然后通过pcie直通给istoreos，已打开移植 Intel GPU 驱动 (i915-oot)选项，通过istoreos安装jellyfin插件，无修改镜像，一打开硬解就无法工作
![image](https://github.com/istoreos/istoreos/assets/78064904/5264f761-8035-4c29-a0ab-ac3ec519697e)
![image](https://github.com/istoreos/istoreos/assets/78064904/94491482-d09c-4625-9c29-48afdc868250)
![image](https://github.com/istoreos/istoreos/assets/78064904/1253abb5-05d6-45e4-b93f-cffcfe3fdcd3)
![image](https://github.com/istoreos/istoreos/assets/78064904/51b55cba-c79a-440a-a686-1312cd2778d4)

**Reply by HongminTan** [reply]
补充一下，如果直通的是非虚拟核显就可以硬解
![image](https://github.com/istoreos/istoreos/assets/78064904/6fa4703a-2442-4512-baa5-c930297f8a43)

**Reply by HongminTan** [reply]
![image](https://github.com/istoreos/istoreos/assets/78064904/a5373145-f84e-41e5-ac8b-37be9359fccc)

**Reply by jjm2473** [reply]
可能是驱动本身不完善。你看看jellyfin的转码日志，还有转码失败以后再看看内核日志有没有报错。

**Comment by zlphoenix** [comment]
有两个问题求助一下：
1.USB 2.5G rtl8156B网卡的问题。iStore固件版本：iStoreOS 22.03.6 2024061415，系统安装在ESXI里面，USB直通。同时板载intel82583网卡也是直通方式。
安装网卡之后偶尔降速到1.3GBps，于是参考网上文章开巨帧，手工安装驱动 kmod-usb-net-rtl8152 ，之后通过这个USB网卡经常掉线，于是使用iperf3 测试发现确实有掉线的情况。请问有什么办法恢复？

2.使用系统便利工具 试图恢复系统文件时出错，iStore菜单无法打开，提示500错误，有什么办法在不丢失现有工具和配置的情况下恢复系统功能吗？非常感谢，错误日志如下：
/usr/lib/lua/luci/template.lua:97: Failed to execute template 'store/main'.
A runtime error occurred: /usr/lib/lua/luci/template.lua:74: Failed to load template 'tasks/embed'.
Error while parsing template '/usr/lib/lua/luci/view/tasks/embed.htm':
No such file or directory
stack traceback:
	[C]: in function 'error'
	/usr/lib/lua/luci/template.lua:74: in function '__init__'
	/usr/lib/lua/luci/util.lua:65: in function 'Template'
	/usr/lib/lua/luci/dispatcher.lua:699: in function 'include'
	[string "/usr/lib/lua/luci/view/store/main.htm"]:3: in main chunk
stack traceback:
	[C]: in function 'error'
	/usr/lib/lua/luci/template.lua:97: in function 
	(tail call): ?
	/usr/lib/lua/luci/controller/store.lua:151: in function
